src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,10 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+
```

Comment:
Since now the storage is only updated when a command is mutation, the header comment should be updated accordingly to reflect the change.